src/pages/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../components/Button/Button';
+import ButtonInactive from '../../components/ButtonInactive/ButtonInactive';
+import {
+  SearchBarForm,
+  Heading,
+  SearchBarContainer,
+  Prefix,
+  Input,
+} from './SearchBar.style';
+
+function SearchBar(props) {
+  // Get the subreddit value from the URL
+  const [searchTerm, setSearchTerm] = useState((new URLSearchParams(window.location.search).get('subreddit')).slice(0, 21));
+  const [isValidSearch, setIsValidSearch] = useState(true);
+  const { handleSearch } = props;
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (isValidSearch) {
+      handleSearch(searchTerm);
+    }
+  }
+
+  function handleChange(event) {
+    // Check if last letter of input is valid
+    const { value } = event.currentTarget;
+    const regexValidInput = /\w+/;
```

Comment:
Thanks for the comment in the other conversation, understood what the regexp is supposed to do.

The problem with checking the last letter only is that it's buggy. Here I enter a value, navigate to the left and add some special characters. When I navigate back to the end of the text and try to delete the characters I'm stuck at the char before a special char.

![image](https://user-images.githubusercontent.com/9722792/78293394-c9065b00-7528-11ea-84a1-c40e204405dc.png)

![image](https://user-images.githubusercontent.com/9722792/78293413-d0c5ff80-7528-11ea-9da5-d08fea98311e.png)

Validation is always tricky and you need to be very careful. Otherwise, you might forbid people for example to signup with an email address that is falsely marked as invalid